controllers/secretutils/secret-utils.go

```diff
@@ -0,0 +1,158 @@
+package secretutils
```

Comment:
You can rename it as secrets.go and have the parent directory as `utils`
Currently `secretutils/secret-utils.go` sounds redundant

Reply:
Majority wins... (=